Tasks/Common/dotNetCliWrapper.ts

```diff
@@ -4,28 +4,119 @@
  */
 
 import tl = require('vsts-task-lib/task')
-import tr = require('vsts-task-lib/toolrunner')
+
+import * as fs from 'fs'
+import * as path from 'path'
+import { ToolRunner } from 'vsts-task-lib/toolrunner'
 
 export class DotNetCliWrapper {
-    public constructor(private readonly cwd: string, private readonly env: any) {}
+    private constructor(
+        private readonly cwd: string,
+        private readonly env: any,
+        private readonly dotnetCliPath: string
+    ) {}
+
+    public async restore(): Promise<number> {
+        return await this.execute(['restore'], '')
+    }
+
+    private async execute(args: string[], additionalArgs: string, additionalCommandLineOptions?: any): Promise<number> {
+        const dotnet = tl.tool(this.dotnetCliPath)
+
+        for (const arg of args) {
+            dotnet.arg(arg)
+        }
+
+        dotnet.line(additionalArgs)
+
+        const execOptions: any = {
+            cwd: this.cwd,
+            env: this.env
+        }
+
+        if (additionalCommandLineOptions) {
+            // tslint:disable-next-line: no-unsafe-any
+            for (const key of Object.keys(additionalCommandLineOptions)) {
+                // tslint:disable-next-line: no-unsafe-any
+                execOptions.key = additionalCommandLineOptions[key]
+            }
+        }
 
-    public async restoreAsync(): Promise<void> {
-        return this.executeAsync(['restore'], undefined)
+        // tslint:disable-next-line: no-unsafe-any
+        return await dotnet.exec(execOptions)
     }
 
-    public async serverlessDeployAsync(
+    private async updateGlobalTools(): Promise<boolean> {
```

Comment:
I would add logging statements when install or update fail. Right now the when an update is done the log statements are going to shoe failure to install the tools then silently fix itself with an update. That could leave users confused.

Reply:
There are, the debug statements are in this block

Reply:
There are only debug statements for failure. I don't see any for when things work correctly. So you will see the debug statetement on the install failure but not the debug statement that update was then successful.

Reply:
Those log statements are in LambdaNETCoreDeploy TaskOperations.ts